src/components/playback/playbackmanager.js

```diff
@@ -2910,9 +3017,15 @@ class PlaybackManager {
             if (mediaSource) {
                 playerData.audioStreamIndex = mediaSource.DefaultAudioStreamIndex;
                 playerData.subtitleStreamIndex = mediaSource.DefaultSubtitleStreamIndex;
+                if (self.trackHasSecondarySubtitleSupport(mediaSource.MediaStreams[mediaSource.DefaultSubtitleStreamIndex])) {
```

Comment:
That would definitely clean things up. What about if the `rankStreamType` best choice happens to be an unsupported secondary stream option? It's unlikely to get enough points because it would not get any points for the matching codec, but it's still possible. 
If it chooses an unsupported track, then we would get the same issue where it will not show in the player, but the index would still be saved in the player state.

https://github.com/jellyfin/jellyfin-web/blob/e1521f9a125c38444afecfe089cbb2140e1821ab/src/components/playback/playbackmanager.js#L2368

Maybe you are correct in that maybe we should consider this as work for a separate PR

Reply:
> What about if the `rankStreamType` best choice happens to be an unsupported secondary stream option? It's unlikely to get enough points because it would not get any points for the matching codec, but it's still possible.

The proposed code checks such a situation.
`rankStreamType` and `autoSetNextTracks` are private, so it is safe to do it not in them.